micromanager_gui/main_window.py

```diff
@@ -148,92 +159,121 @@ def add_frame_explorer(self, name, array):
         except KeyError:
             self.viewer.add_image(array, name=layer_name)
 
-    # TO DO: add the file name form the save box
-    def _on_mda_frame(self, image, event):
 
-        sequence = event.sequence
+    # create temp folder when mda starts
+    def _on_mda_started_temp_folder(self, sequence: useq.MDASequence):
 
-        event.index.setdefault('t', 0)
-        event.index.setdefault('z', 0)
+        self.temp_folder = tempfile.TemporaryDirectory(None, str(sequence.uid))
+        return self.temp_folder.name
 
-        t_stack_length = len(sequence.time_plan) or 1
-        p_stack_length = len(sequence.stage_positions)
-        z_stack_length = len(sequence.z_plan) or 1
-        c_stack_length = len(sequence.channels)
+    #get the name to use to save the active ayer
+    def get_filename(self, fname, list_dir):
+        val = int(fname.split('_')[-1])
+        while True:
+            new_val = '{0:03}'.format(val)
+            fname = fname[:-3] + new_val
+            if ((fname + '.tif') or (fname + '.tiff')) not in set(list_dir):
+                break
+            else:
+                val += 1
+        return fname
 
-        event_index_t = event.index["t"]
-        event_index_p = event.index["p"]
-        event_index_z = event.index["z"]
-        event_index_c = event.index["c"]
-        
-        file_name = 'mda'
+    # delete temp folder and files when mda is done and save layer
+    # and add increment to save name
+    def _on_mda_finished_save_layer(self, sequence: useq.MDASequence):
+            
+        if self.mda.save_groupBox.isChecked():
 
-        layer_name = (
-            f'{file_name}_[{sequence.axis_order}]_p{p_stack_length}_'
-            f't{t_stack_length}_z{z_stack_length}_c{c_stack_length}'
-        )
+            active_layer = self.viewer.layers.selection.active.name
 
-        try:
-            layer = self.viewer.layers[layer_name]
+            fname =  self.mda.fname_lineEdit.text()
 
-            if sequence.axis_order == 'tpzc' or sequence.axis_order == 'ptzc':            
-                #channels
-                if event_index_c > 0 and event_index_z == 0 and event_index_p == 0 and event_index_t == 0:
-                    empty_im = np.empty(((1,)*len(sequence.axis_order) + image.shape), dtype=np.uint16) 
-                    layer.data = np.concatenate((layer.data,empty_im), axis=-3)
+            save_path = Path(self.mda.dir_lineEdit.text())
+
+            list_dir = [str(i).split('/')[-1] for i in save_path.iterdir() \
+                if (str(i).split('/')[-1]).endswith(('.tif', '.tiff'))]
+
+            try:
```

Comment:
can you put some comments in here to make it easier for me to quickly determine what you're trying to accomplish?  (see comments in `_on_mda_frame` for an example of what I mean)